doc/howtos/authorized-keys.rst

```diff
@@ -0,0 +1,104 @@
+
+Generating a Private/Public Key Pair
+====================================
+
+On Linux systems (or Windows systems with `Git Bash`_ installed) use the ``ssh-keygen`` utility to generate
+an ssh key pair by following its instructions ::
+
+   $ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+
+   # Creates a new ssh key using the provided email as a label
+   Generating public/private rsa key pair.
+   Enter a file in which to save the key ($HOME/.ssh/id_rsa): [Press enter]
+   # Press enter to accept the default file names or provide a name for development use, for example
+
+   Enter passphrase (empty for no passphrase): [Type a passphrase]
+   # If you're only using this key for local development work, you could skip a passphrase and press Enter
+   Enter same passphrase again: [Type passphrase again]
+
+At this point the tool will generate an ssh key pair and store the public and private keys in separate files in the ``$HOME/.ssh/`` folder.
+
+On Windows systems, you can also use **PuTTYgen**.  If needed, download PuTTYgen from the `PuTTY download page`_ (you may already have
+it installed on your Windows system).  Launch the program and click the **Generate** button.  (If you're only using this key for 
+local development work, you could skip a passphrase.) Save the key-pair files by clicking the **Save public key** and **Save private key** buttons.  
+
+You can read more information about key generation and use of pass phrases from GitHub's `generating an ssh key help`_ and other online sources.
+
+.. _`PuTTY download page`: http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html
+.. _`Git Bash`: https://git-for-windows.github.io/
+.. _`generating an ssh key help`: https://help.github.com/articles/generating-an-ssh-key/
+
+
+Build-Time Configuration of Authorized Keys
+=========================================== 
+
+The recommended approach for enabling remote ``ssh`` access is to install authorized public ssh keys by 
+setting OSTRO_ROOT_AUTHORIZED_KEYS in your ``local.conf`` configuration file and building them into your image.
+
+#. Look in your ``local.conf`` file and find the line:::
+
+   # OSTRO_ROOT_AUTHORIZED_KEYS = "ssh-rsa AAA...== john@example.com\nssh-dss AA...FPaQ== joan@example.com"
+
+   Uncomment it, and replace the quoted string with the contents of the public key file(s) you'd like to authorize
+   for remote access.  Separate
+   multiple public keys with "\\n", as in the example shown. 
+   The contents of this configuration key will be placed in ``~root/.ssh/authorized_keys`` when the image is built.
+
+#. Make note of your devices ip address using ``ifconfig`` for example, if you've got a local console or serial port connection.
+
+#. Login from your host using ``ssh`` with your private key (from host) and the ip address of your device, for example: ::
+
+    $ ssh root@192.168.2.2 -i $HOME/.ssh/id_rsa
+
+  Since root does not have a password, you won't be asked for one and you'll be logged in.
+
```

Comment:
here probably also line that in putty, user should be in put in Connection->ssh->auth path to the private key file